### PR TITLE
fix(dynamic-forms): preserve nested array-item defaults on partial value

### PR DESCRIPTION
**Pull request by dereekb**
# Pull Request

## Description

Follow up to #387. Found an issue with nested arrays where when a value was set and that value had fields that were undefined then those fields would not render. Below is a screenshot of the same array, with the array's value looked like this:

```json
"list": [
    {
      "toggle": true,
      "toggleWrapperLabel": false
    },
    {
      "toggle": "",
      "toggleWrapperLabel": "",
      "toggleBothLabel": "",
      "toggleContentLabel": ""
    }
  ]
```

Screenshot 2026-05-05 at 8 09 01 PM

## Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] 🎨 Code style update (formatting, renaming)
- [ ] ♻️ Code refactoring (no functional changes)
- [ ] ⚡ Performance improvement
- [ ] ✅ Test update
- [ ] 🔧 Build/tooling update

## Related Issues

Fixes #
Related to #

## Changes Made

-Extend deepMergeDefaults to merge object array elements positionally so a partial nested object inside an array (e.g. an item missing one of its declared sub-field keys) no longer triggers NG01904: Orphan field from the Signal Forms validation graph. Mirrors the prior group-field fix.

## Testing

- [x] Unit tests added/updated
- [x] All tests pass locally
- [ ] Tested manually
- [ ] Tested edge cases

## Documentation

- [ ] Documentation updated (if needed)
- [ ] API documentation (TSDoc) added/updated
- [ ] Examples updated (if needed)

## Checklist

- [x] Code follows the [coding standards](../CODING_STANDARDS.md)
- [x] No linting errors (`pnpm lint`)
- [x] Code is formatted (`pnpm lint:fix`)
- [x] Build succeeds (`pnpm build:libs`)
- [x] Commits follow [conventional commit format](https://www.conventionalcommits.org/)
- [x] Self-review completed
- [x] No commented-out code or console.log() statements
- [x] Type safety maintained (no `any` types)

### CODE REVIEW

**Comment by dereekb** [comment]
CI E2E Tests caught some issues with the fix.

Summary of the fix:                                                                                                                                                                                                                               
  - mergeArrayDefaults now only merges positionally when defaults.length === value.length. When lengths differ — every runtime add/insert/remove — it returns the value array reference as-is.                                                    
  - This preserves the original group-level fix (length-matching declared arrays with partial items still fill in missing sub-field keys) while restoring pre-fix reference identity for runtime array ops, so Signal Forms' item bindings stay     
  aligned with rendered DOM rows.                                                                                                                                                                                                                 
  - Added a regression test that asserts result.items is the same reference as the input valueArr when lengths differ (i.e. the prepend case).

File: packages/dynamic-forms/src/lib/utils/object-utils.ts [hunk 1]
```diff
@@ -250,6 +267,32 @@ export function deepMergeDefaults<T extends Record<string, unknown>>(defaults: T
...
+ * desync item FieldTrees from the rendered DOM rows.
+ *
+ * When lengths match (the declared/initial-value path the array fix targets)
+ * each `defaults[i]`/`value[i]` pair is deep-merged when both are plain
+ * objects so partial array-item values don't drop sibling sub-field keys.
+ */
+function mergeArrayDefaults(defaults: readonly unknown[], value: readonly unknown[]): readonly unknown[] {
+  if (defaults.length !== value.length) return value;
+
+  return value.map((item, i) => {
```
Comment (antimprisacaru) [inline packages/dynamic-forms/src/lib/utils/object-utils.ts:292]: Equal-length primitive arrays (e.g. `tags: ['a', 'b']`) now get a fresh array reference on every call via `.map()`, whereas the old code returned incoming directly. Since Signal Forms uses reference equality for item tracking, this could cause `FieldTree` rebuilds for primitive-array fields. Suggest short-circuiting before the `.map()` when no element pair is a plain object:

```
if (!value.some((item, i) => isPlainObject(defaults[i]) && isPlainObject(item))) return value;
```

File: packages/dynamic-forms/src/lib/utils/object-utils.spec.ts [hunk 2]
```diff
@@ -346,13 +346,52 @@ describe('object-utils', () => {
...
+    it('should merge object array elements positionally so partial items keep declared sub-field defaults', () => {
+      const defaults = { items: [{ checkboxA: false, checkboxB: false }] };
+      const value = { items: [{ checkboxA: true }] };
+
+      expect(deepMergeDefaults(defaults, value)).toEqual({
+        items: [{ checkboxA: true, checkboxB: false }],
+      });
+    });
+
+    it('should pass through object array elements that have no matching default', () => {
```
Comment (antimprisacaru) [inline packages/dynamic-forms/src/lib/utils/object-utils.spec.ts]: This name describes the wrong thing — the condition is actually a length mismatch (value has 2 items, defaults has 1), not a missing default. Suggest: 'should return value as-is when value has more items than defaults'. Also worth adding a companion test for equal-length primitive arrays asserting result.tags === valueArr (which would currently fail due to the .map() issue above).

**Review by antimprisacaru — COMMENTED** [review]
Nice find! Thanks for contribution! Left a couple small comments :)

**Review by antimprisacaru — APPROVED** [review]
